### ISSUE TITLE
[BUG] smartctl returned an error code (64) while processing sda\n" type=metrics

### ISSUE THREAD

**Issue by ovizii** [AnalogJ/scrutiny#286]
**Describe the bug**
I see strange values reported by scrutiny i.e. for "power on hours" for SATA disks, values are shown just fine for nvmes

**Expected behavior**
I was expecting to see the correct value i.e. 55413 hours

**Screenshots**
![image](https://user-images.githubusercontent.com/837902/172336943-8d96318c-e302-4a0d-871c-0bd818cc30d2.png)

![image](https://user-images.githubusercontent.com/837902/172337133-8baf82d6-b7f5-4b90-bbe0-8bd609f35334.png)

**Log Files**
scrutiny    | AnalogJ/scrutiny/metrics                                dev-0.4.9                                                                      
scrutiny    |                                                                                                                                        
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Verifying required tools" type=metrics                                                
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --scan --json" type=metrics                               
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json --device nvme /dev/nvme1" type=metrics      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Using WWN Fallback" type=metrics                                                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sda" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sdb" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sdc" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sdd" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sde" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json /dev/sdf" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Generating WWN" type=metrics                                                          
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --info --json --device nvme /dev/nvme0" type=metrics      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Using WWN Fallback" type=metrics                                                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Sending detected devices to API, for filtering & validation" type=metrics             
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="127.0.0.1 - scrutinity [07/Jun/2022:10:45:05 +0200] \"POST /api/devices/register\" 200 4467 \"\" \"Go-http-client/1.1\" (2ms)" clientIP=127.0.0.1 hostname=scrutinity latency=2 method=POST path=/api/devices/register referer= respLength=4467 statusCode=200 userAgent=Go-http-client/1.1                                                                                                     
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Collecting smartctl results for nvme1\n" type=metrics                                 
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --xall --json --device nvme /dev/nvme1" type=metrics      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Publishing smartctl results for 203e079c191500049128\n" type=metrics                  
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="127.0.0.1 - scrutinity [07/Jun/2022:10:45:05 +0200] \"POST /api/device/203e079c191500049128/smart\" 200 16 \"\" \"Go-http-client/1.1\" (45ms)" clientIP=127.0.0.1 hostname=scrutinity latency=45 method=POST path=/api/device/203e079c191500049128/smart referer= respLength=16 statusCode=200 userAgent=Go-http-client/1.1                                                                     
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Collecting smartctl results for sda\n" type=metrics                                   
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Executing command: smartctl --xall --json /dev/sda" type=metrics                      
scrutiny    | time="2022-06-07T10:45:05+02:00" level=error msg="smartctl returned an error code (64) while processing sda\n" type=metrics            
scrutiny    | time="2022-06-07T10:45:05+02:00" level=error msg="smartctl detected a error log with errors" type=metrics                              
scrutiny    | time="2022-06-07T10:45:05+02:00" level=info msg="Publishing smartctl results for 0x50014ee6b11492e7\n" type=metrics      
```


manually doing : smartctl -d ata --all /dev/sdf  

I see:
9 Power_On_Hours          0x0032   030   030   000    Old_age   Always       -       51433 

and the raw value of 51433 seems correct as it translates to roughly 5.8 years which is correct.

If needed, I can try to get you more debug log files, but the process looked complicated. I will have to read the instructions again and try to figure out how to get you the debug data.

**Comment by AnalogJ** [comment]
Hey @ovizii 
The exit code (64) from smartctl is not somethign that I can control, but you can see what they mean by looking at the table here: https://github.com/AnalogJ/scrutiny/blob/master/docs/TROUBLESHOOTING_DEVICE_COLLECTOR.md#exit-codes

Regarding the Power_On_Hours mismatch, that could be due to the normalized value being compared to the raw value for ata drives vs nvme drives. I'd have to look at the code to confirm. If you hover over the number `42` in the Value column, it should tell you the value type. 

While your confusion is understandable, the UI for displaying this data will be changing soon. This shouldnt be an issue for much longer.

**Comment by ovizii** [comment]
Thanks for explaining. I guess I'll have to live with it as is or watch for updates in case the UI changes. :-) 

No worries though, I just found it weird just seeing some random number like 42 when I knew exactly how old my HDs were. Also seeing a number like 42 doesn't really help a human looking at scrutiny at all. Not saying that 19201 is much more helpful. :-)

Feel free to close this, I simply wanted to report it since I saw this issue which to me sounded like this should be fixed in 0.4.x versions of scrutiny => https://github.com/AnalogJ/scrutiny/issues/199
